include/cglm/types.h

```diff
@@ -45,6 +46,30 @@ typedef CGLM_ALIGN_IF(32) vec4  mat4[4];
 typedef CGLM_ALIGN_IF(16) vec4  mat4[4];
 #endif
 
+// struct types
+typedef union CGLM_ALIGN_IF(16) vec3s {
+#ifndef CGLM_NO_ANONYMOUS_STRUCT
+  struct {
+    float x;
+    float y;
+    float z;
+  };
+#endif
+  vec3 raw;
+} vec3s;
+
+typedef union CGLM_ALIGN_IF(16) vec4s {
+#ifndef CGLM_NO_ANONYMOUS_STRUCT
```

Comment:
from here: https://gcc.gnu.org/onlinedocs/gcc-4.0.2/gcc/Type-Attributes.html
> Note that the alignment of any given struct or union type is required by the ISO C standard to be at least a perfect multiple of the lowest common multiple of the alignments of all of the members of the struct or union in question. This means that you can effectively adjust the alignment of a struct or union type by attaching an aligned attribute to any one of the members of such a type...

So, I think we can drop the alignment on the struct and leave the alignment of the `raw` field since those will have offset `0` anyways, it will force the whole struct to be aligned as specified by that field, right?

Reply:
Sounds good, I guess other compilers e.g. clang, msvc... should work same way but I'm not sure. Let's keep same alignment of raw on union as it is (like you did).